ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -6,14 +6,81 @@ set -o pipefail
 
 echo "************ baremetalds packet setup command ************"
 
+cd
+cat > packet-setup.yaml <<-EOF
+- name: setup Packet host
+  hosts: localhost
+  collections:
+   - community.general
+  gather_facts: no
+  vars:
+    - cluster_type: "{{ lookup('env', 'CLUSTER_TYPE') }}"
+    - slackhook_path: "{{ lookup('env', 'CLUSTER_PROFILE_DIR') }}/slackhook"
+    - packet_project_id: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-project-id') }}"
+    - packet_auth_token: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-auth-token') }}"
+  tasks:
+  - name: write fix uid file
+    copy:
+      content: |
+        # Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+        # to be able to SSH.
+        if ! whoami &> /dev/null; then
+            if [ -x "\$(command -v nss_wrapper.pl)" ]; then
+                grep -v -e ^default -e ^\$(id -u) /etc/passwd > "/tmp/passwd"
+                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/tmp/passwd"
+                export LD_PRELOAD=libnss_wrapper.so
+                export NSS_WRAPPER_PASSWD=/tmp/passwd
+                export NSS_WRAPPER_GROUP=/etc/group
+            elif [[ -w /etc/passwd ]]; then
+                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
+            else
+                echo "No nss wrapper, /etc/passwd is not writeable, and user matching this uid is not found."
+                exit 1
+            fi
+        fi
+      dest: "${SHARED_DIR}/fix-uid.sh"
+
+  - name: write Packet common configuration file
+    copy:
+      content: |
+        source "\${SHARED_DIR}/fix-uid.sh"
+
+        # Initial check
+        if [[ ! "\${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]]; then
+            echo >&2 "Unsupported cluster type '\${CLUSTER_TYPE}'"
+            exit 1
+        fi
+
+        IP=\$(cat "\${SHARED_DIR}/server-ip")
+        SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -o LogLevel=ERROR -i "\${CLUSTER_PROFILE_DIR}/packet-ssh-key")
+
+        # Checkout packet server
+        for x in \$(seq 10) ; do
+            test "\${x}" -eq 10 && exit 1
+            ssh "\${SSHOPTS[@]}" "root@\${IP}" hostname && break
+            sleep 10
+        done
+      dest: "${SHARED_DIR}/packet-conf.sh"
+EOF
+ansible-playbook packet-setup.yaml -e "packet_hostname=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"  |& gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush(); }'
+
+if [ $CLUSTER_TYPE == "equinix-ocp-metal" ] && [ -e "${CLUSTER_PROFILE_DIR}/ofcir_url" ] ; then
```

Comment:
Isn't redundant the `CLUSTER_TYPE` check now? As long as the `ofcir_url` file is available it should be sufficient to trigger the alternative execution.
It could also useful to add an echo to trace that a different approach is taken

Reply:
removed the CLUSTER_TYPE check and added a bit more logging